plugin-server/tests/main/db.test.ts

```diff
@@ -176,6 +177,91 @@ describe('DB', () => {
+            expect(fetchedPerson).toEqual(undefined)
+        })
+
+        describe('clickhouse behavior', () => {
+            beforeEach(async () => {
+                await resetTestDatabaseClickhouse()
+                // :TRICKY: Avoid collapsing rows before we are able to read them in the below tests.
+                await db.clickhouseQuery('SYSTEM STOP MERGES')
+            })
+
+            afterEach(async () => {
+                await db.clickhouseQuery('SYSTEM START MERGES')
+            })
+
+            async function fetchPersonsRows(options: { final?: boolean } = {}) {
+                const query = `SELECT * FROM person ${options.final ? 'FINAL' : ''}`
+                return (await db.clickhouseQuery(query)).data
+            }
+
+            it('marks person as deleted in clickhouse', async () => {
+                const team = await getFirstTeam(hub)
+                // :TRICKY: We explicitly don't create distinct_ids here to keep the deletion process simpler.
+                const person = await db.createPerson(TIMESTAMP, {}, {}, {}, team.id, null, true, uuid, [])
+                await delayUntilEventIngested(fetchPersonsRows, 1)
+
+                // We do an update to verify
+                await db.updatePersonDeprecated(person, { properties: { foo: 'bar' } })
+                await db.kafkaProducer.flush()
+                await delayUntilEventIngested(fetchPersonsRows, 2)
+
+                const kafkaMessages = await db.deletePerson(person)
+                await db.kafkaProducer.queueMessages(kafkaMessages)
+                await db.kafkaProducer.flush()
+
+                const persons = await delayUntilEventIngested(fetchPersonsRows, 3)
+
+                expect(persons).toEqual(
+                    expect.arrayContaining([
+                        expect.objectContaining({
+                            id: uuid,
+                            properties: JSON.stringify({}),
+                            is_deleted: 0,
+                            version: 0,
+                        }),
+                        expect.objectContaining({
+                            id: uuid,
+                            properties: JSON.stringify({ foo: 'bar' }),
+                            is_deleted: 0,
+                            version: 1,
+                        }),
+                        expect.objectContaining({
+                            id: uuid,
+                            is_deleted: 1,
+                            version: 2,
+                        }),
+                    ])
+                )
+
+                const personsFinal = await fetchPersonsRows({ final: true })
+                expect(personsFinal).toEqual(
+                    expect.arrayContaining([
+                        expect.objectContaining({
+                            id: uuid,
+                            is_deleted: 1,
+                            // :TODO: This is wrong
```

Comment:
RE my previous comment I guess the idea is to fix this in a follow up?

Reply:
Dead comment, sorry. Should have removed this in the last commit.